manifests/opsmanager/install.pp

```diff
@@ -0,0 +1,48 @@
+# PRIVATE CLASS: do not call directly
+class mongodb::opsmanager::install {
+  #assert_private("You are calling a private class mongodb::opsmanager::install.")
+  $package_ensure = $mongodb::opsmanager::package_ensure
+  $package_name   = $mongodb::opsmanager::package_name
+  $download_url   = $mongodb::opsmanager::download_url
+
+  case $package_ensure {
+    true:     {
+      $my_package_ensure = 'present'
+      $file_ensure       = 'directory'
+    }
+    false:    {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'absent': {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    'purged': {
+      $my_package_ensure = 'purged'
+      $file_ensure       = 'absent'
+    }
+    default:  {
+      $my_package_ensure = $package_ensure
+      $file_ensure       = 'present'
+    }
+  }
+
+  case $facts['os']['family'] {
+    'RedHat': {
+      $my_provider = 'rpm'
+    }
+    default: {
+      warning("The ${module_name} module might not work on ${facts['os']['family']}.  Sensible defaults will be attempted.")
+      $my_provider = undef
+    }
+  }
+
+  Package { provider => $my_provider }
+
+  package { $package_name:
+    ensure   => $my_package_ensure,
+    source   => $download_url,
+    provider => $my_provider,
```

Comment:
Are you sure that it is required to set the provider? Puppet should be able to figure this out by itself. Than we could get rid of the above case statement.

Reply:
This is needed to support `source`.  eg you need to use `rpm` instead of `yum`.

Reply:
`yum` provider can use `source` since puppet 5.4.0 only.  See https://tickets.puppetlabs.com/browse/PUP-3323

Reply:
I did some tests. Puppet 4 uses the yum provider by default, even if `source` is set. It will fail to install the rpm file. So we need to set the provider, at least for puppet 4.